### DISCUSSION TITLE
Quick Launch toolbar replacement

### DISCUSSION THREAD

**Discussion by valinet** [valinet/ExplorerPatcher#819] [Announcements]
Hi

As I have come to relatively recently learn, a good bunch of the people still using (like myself) the venerable legacy taskbar also have a preference for a toolbar which they can use for quickly launching applications.

Before the superbar was introduced in Windows 7, this task was traditionally accomplished by using the "Quick Launch" toolbar. Since Windows 7, the officially sanctioned way is for users to pin items to the taskbar, although, as many of you know, it's not an exact replacement. Unfortunately, although still there, the deskband infrastructure was kind of deprecated, it never receiving modern features like support for DPI scaling changes, proper alignment of the icons in certain modes etc.

The idea that I am presenting today is not necessarily a direct 1:1 replacement either, but rather my take on how to tackle this situation based on what is currently available in the existing legacy taskbar. I implemented this recently, as of ExplorerPatcher pre-release version 22000.469.42.7, as I found myself needing as well something quicker to launch some apps, without needing to hold the `Shift` key, right clicking etc.

There are 2 new options in the Advanced section of the app:

* Pinned items act as quick launch (don't group pinned items with active apps) * - this will make it so that the apps that you launch do not group together with the pinned button of the app, if any; thus, pinned apps always stay on the left, same as the traditional quick launch toolbar used to do
* When the taskbar shows button labels, remove the extra gap around pinned items - this will remove the extra gap between the pinned items, so that they take a less bit space

The look is similar to this:

![image](https://user-images.githubusercontent.com/6503598/153023524-15768765-fbd7-4aae-a4af-d6eef2be05cf.png)

The advantage with this method is that it gets full support for high DPI scaling, and the icons also display properly, in-line with the rest of the taskbar icons. The disadvantages, are obviously, being unable to place it wherever on the taskbar, right click menu being the jump lists instead of the shell item context menu, the inability to hide some of the items under a chevron (although that may be addressed with a custom app that acts like a launcher, if someone implements that) etc.

These features are already available in the very robust and popular [7+ Taskbar Tweaker](https://tweaker.ramensoftware.com/) app which  I highly recommend. Since the app is mostly open source and with a compatible license, I also implemented those features in ExplorerPatcher so that I do not have to run 2 apps separately. This doesn't mean that EP will in the future replicate all that's in 7+TT; that is not and will never be on the roadmap, the utility is recommended if you need other functionalities that it offers.

Thanks, and I hope it helps some of you.

**Comment by l0oo** [comment]
There is also possibility to bring back old good quick launch to W10 taskbar:
https://www.lifewire.com/add-quick-launch-toolbar-in-windows-10-5115231

**Reply by plofhaan** [reply]
Good tutorial, but as an added note, in Step 3 the folder location doesn't necessarily need to be `%APPDATA%\Microsoft\Internet Explorer\Quick Launch\` because it also is possible to point to a different folder instead (which is how I have been doing it on all versions of Windows ever since Win7 was officially released, i.e., just copy the Show Desktop shortcut from that folder over to your own folder that you have created, like, e.g., `%USERPROFILE\Quick Launch`). Further, in Step 7 it also is possible to click on the vertical line that’s located to the left of the quick launch toolbar, and drag it all the way to the left, to get the same result. (On Win7 you needed to drag this vertical line really fast to make it 'snap', else it kept refusing to let itself be moved past the 'Pinned' area so, basically, people who were too slow never got to enjoy quick launch on Win7, lol...)

One of the things I like about this 'classic' method to bring back the Quick Launch is that, compared to _pinned_ icons, the icons on a _toolbar_ are smaller and spaced closer together if Scaling is set to 200% on a 4k screen. As a result, a toolbar takes up noticeably less space than a bunch of pinned icons. In addition to this, I still find toolbars easier to visibly distinguish from active apps than pinned icons even with this new feature of EP. I mean, there is no separator shown between the pinned icons and the active apps, and, although I guess I could always decide to just pin a dummy icon in that position to work my way around the issue, doing so would take up even _more_ space still.

Pinned icons being always on the left can become another annoyance at times when the number of pinned icons needs to be changed. That's just because changing the number of pinned icons causes the icon positions of the active apps to be changed also. Whereas changing the visible contents of a toolbar does not. This matters to me mainly because the contents of my toolbar (that I have positioned in the center of my taskbar) can be changed for me automatically during some tasks/actions that can be a part of my workflow, through a few simple scripts that I wrote.

Pinned icons do give the extra benefit of Jump Lists, though. That in fact is one reason why File Explorer is the only thing that I have pinned to my taskbar. Its icon is always on the left because it's the only one that I have pinned, therefore the grouping causes no unwanted visual distraction to me. With the shortcuts that I keep on my toolbar, occasionally I still wish they also had Jump Lists on them. Luckily, though, Open-Shell brings a similar kind of functionality to the Start Menu. Even so, if it could be possible for EP to have an option to add this same or similar functionality directly to the toolbar's icons, and also make the toolbars more customizable, similar to how customization works with Classic Explorer bar (i.e. the separate feature of Open-Shell that applies to File Explorer), I think I can see how that would be like another dream come true. :-)

**Comment by donkzquixote** [comment]
"Thanks, and I hope it helps some of you."
It does, so thanks! :-)
One drawback of the similar feature in 7+ Taskbar Tweaker was that every time the explorer.exe process was restarted, all the taskbar items would be resorted and all the open apps would be placed next to their pinned button again. ExplorerPatcher doesn't do that. :-)

**Comment by Garryb123** [comment]
I've been putting off buying a new Windows desktop because of the horrible W11 Taskbar but after finding your ExplorerPatcher and testing it (and Open-Shell) on my W11 laptop I realize between the 2 programs the Start menu and Taskbar function very well. Thank you for building the program. Please keep it up. I'll be buying it soon.

**Reply by pyrates999** [reply]
EP is free.  Perhaps you were talking about making a donation?

**Reply by Garryb123** [reply]
Thank you. I'm very new to ExplorerPatcher and was mistaken. Yes, I made a donation. I'm very thankful to the developer for making this software.
